File: 1233-remove-sub-folders-from-the-filesystem/1233-remove-sub-folders-from-the-filesystem.js
/**
 * @param {string[]} folder
 * @return {string[]}
 */
var removeSubfolders = function(folder) {
    folder = folder.sort((a,b)=>a.length - b.length);
    const set = new Set();
    for(let val of folder){
        const arr = [...set];
        let flag = true;
        for(let i=0; i<arr.length; i++){
            if(val.startsWith(arr[i]+"/")){
                flag = false
                break;
            }
        }
        if(flag) set.add(val);
    }
    return [...set]
};

File: 1233-remove-sub-folders-from-the-filesystem/1233-remove-sub-folders-from-the-filesystem.java
class Solution {
    public List<String> removeSubfolders(String[] folder) {
        Arrays.sort(folder);

        List<String> res = new ArrayList<>();
        String prev = "";
        for(String path : folder){
            if(prev.isEmpty() || !path.startsWith(prev+"/")){
                res.add(path);
                prev = path;
            }
        }
        return res;
    }
}
